apache/incubator-zeppelin
# ZEPPELIN-543: Storage initialization failure recovery

### What is this PR for?
Handling edge case failures when initializing notebook storage class(es).

### What type of PR is it?
Improvement

### Todos
* [x] - fix repo initialization
* [x] - add different config tests

### Is there a relevant Jira issue?
[ZEPPELIN-543](https://issues.apache.org/jira/browse/ZEPPELIN-543)

### How should this be tested?
There're various test cases (mostly included as unit tests).
One of the test scenarios:
1. in `conf/zeppelin-env.sh` add the following line
    ```
    export ZEPPELIN_NOTEBOOK_STORAGE="org.apache.zeppelin.notebook.repo.VFSNotebookRepo,org.apache.zeppelin.notebook.repo.DummyNotebookRepo"
    ```
2. Start Zeppelin
3. Previously Zeppelin would fail to start, now you would see that only first storage class is initialized and second just didn't initialize.

### Screenshots (if appropriate)

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? No
